narwhals/utils.py

```diff
@@ -42,3 +42,13 @@ def parse_version(version: Sequence[str | int]) -> tuple[int, ...]:
 
 def isinstance_or_issubclass(obj: Any, cls: Any) -> bool:
     return isinstance(obj, cls) or issubclass(obj, cls)
+
+
+def validate_same_library(one_object: Any) -> None:
```

Comment:
maybe this could/should be a bit more specific? `items: Iterable[Any]`?

Reply:
Done! :)